src/gui/DetailsWidget.cpp

```diff
@@ -32,325 +32,320 @@
 #include "gui/DatabaseWidget.h"
 #include "entry/EntryAttachmentsModel.h"
 
+namespace {
+constexpr int GeneralTabIndex = 0;
+}
+
 DetailsWidget::DetailsWidget(QWidget* parent)
     : QWidget(parent)
     , m_ui(new Ui::DetailsWidget())
     , m_locked(false)
     , m_currentEntry(nullptr)
     , m_currentGroup(nullptr)
-    , m_timer(nullptr)
-    , m_attributesTabWidget(nullptr)
-    , m_attachmentsTabWidget(nullptr)
-    , m_autotypeTabWidget(nullptr)
+    , m_step(0)
+    , m_totpTimer(new QTimer(this))
     , m_selectedTabEntry(0)
     , m_selectedTabGroup(0)
 {
     m_ui->setupUi(this);
 
-    connect(parent, SIGNAL(pressedEntry(Entry*)), SLOT(getSelectedEntry(Entry*)));
-    connect(parent, SIGNAL(pressedGroup(Group*)), SLOT(getSelectedGroup(Group*)));
-    connect(parent, SIGNAL(currentModeChanged(DatabaseWidget::Mode)), SLOT(setDatabaseMode(DatabaseWidget::Mode)));
-
-    m_ui->totpButton->setIcon(filePath()->icon("actions", "chronometer"));
-    m_ui->closeButton->setIcon(filePath()->icon("actions", "dialog-close"));
+    // Entry
+    m_ui->entryTotpButton->setIcon(filePath()->icon("actions", "chronometer"));
+    m_ui->entryCloseButton->setIcon(filePath()->icon("actions", "dialog-close"));
 
-    connect(m_ui->totpButton, SIGNAL(toggled(bool)), SLOT(showTotp(bool)));
-    connect(m_ui->closeButton, SIGNAL(toggled(bool)), SLOT(hideDetails()));
-    connect(m_ui->tabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndex(int)));
+    m_ui->entryAttachmentsWidget->setReadOnly(true);
+    m_ui->entryAttachmentsWidget->setButtonsVisible(false);
 
-    m_ui->attachmentsWidget->setReadOnly(true);
-    m_ui->attachmentsWidget->setButtonsVisible(false);
+    connect(m_ui->entryTotpButton, SIGNAL(toggled(bool)), m_ui->entryTotpWidget, SLOT(setVisible(bool)));
+    connect(m_ui->entryCloseButton, SIGNAL(toggled(bool)), SLOT(hide()));
+    connect(m_ui->entryTabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndexes()), Qt::QueuedConnection);
 
-    m_attributesTabWidget = m_ui->tabWidget->widget(AttributesTab);
-    m_attachmentsTabWidget = m_ui->tabWidget->widget(AttachmentsTab);
-    m_autotypeTabWidget = m_ui->tabWidget->widget(AutotypeTab);
+    connect(m_totpTimer, SIGNAL(timeout()), this, SLOT(updateTotp()));
 
-    this->hide();
+    // Group
+    m_ui->groupCloseButton->setIcon(filePath()->icon("actions", "dialog-close"));
+    connect(m_ui->groupCloseButton, SIGNAL(toggled(bool)), SLOT(hide()));
+    connect(m_ui->groupTabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndexes()), Qt::QueuedConnection);
 }
 
 DetailsWidget::~DetailsWidget()
 {
 }
 
-void DetailsWidget::getSelectedEntry(Entry* selectedEntry)
+void DetailsWidget::setEntry(Entry* selectedEntry)
 {
-    if (!selectedEntry) {
```

Comment:
Why removing this single check and adding checks in every update function?

Also, in theory `m_currentEntry` should never be equal to `nullptr` but in such case I prefer hiding the detailsView instead of having it blank

Reply:
I was planning to make some changes to the logic of the widget displaying (in my opinion it is better that widget was always visible, and the fields cleared when there are no selected items), but it's probably better to discuss this in another PR. I'll revert some changes in the next hour :)